File: Data Structures & Algorithms/count-paths/submission-0.cpp
class Solution {
public:
int rec(int i,int j,int m,int n){
    if(i>=m-1 && j>=n-1) return 1;
    int ans=0;
    if(i==m-1 && j<n-1){
        return rec(i,j+1,m,n);
    } 
    if(i<m-1 && j==n-1){
        return rec(i+1,j,m,n);
    }
    
      else{
                   ans+=rec(i,j+1,m,n);
                    ans+=rec(i+1,j,m,n);
      }

    
    return ans;
}
    int uniquePaths(int m, int n) {
        return rec(0,0,m,n);
        
    }
};

File: Data Structures & Algorithms/reorganize-string/submission-0.py
from heapq import heappush,heappop
from collections import Counter
class Solution:
    def reorganizeString(self, s: str) -> str:
        freq=Counter(s)
        pq=[]
        for c,f in freq.items():
            heappush(pq,(-f,c))
        
        res=[]
        prev_count=0
        prev_char=""
        while pq :
            cnt,ch=heappop(pq)
            res.append(ch)
            if prev_count < 0 :
                heappush(pq,(prev_count,prev_char))
            
            cnt+=1
            prev_count=cnt
            prev_char = ch
        
        ans="".join(res)

        return ans if len(ans) == len(s) else ""
        



       



        